app/questionnaire/navigator.py

```diff
@@ -33,14 +33,41 @@ def evaluate_goto(goto_rule, answers, group_instance):
     :return:
     """
     if 'when' in goto_rule.keys():
-        answer_index = goto_rule['when']['id']
-        filtered = answers.filter(answer_id=answer_index, group_instance=group_instance)
-        if len(filtered) == 1:
-            return evaluate_rule(goto_rule, filtered[0]['value'])
+
+        if 'id' in goto_rule['when']:
+            answer_index = goto_rule['when']['id']
+            filtered = answers.filter(answer_id=answer_index, group_instance=group_instance)
+            if len(filtered) == 1:
+                return evaluate_rule(goto_rule, filtered[0]['value'])
+
+        if 'meta' in goto_rule['when']:
+            meta_index = goto_rule['when']['meta']
+            if _contains_in_dict(metadata, meta_index):
+                metadata_value = _get_from_dict(metadata, meta_index)
+                return evaluate_rule(goto_rule, metadata_value)
+
         return False
     return True
 
 
+def _get_from_dict(d, keys):
+    if "." in keys:
```

Comment:
Why have we got the dot separated variables i.e. variant.sexual_identity? Can it just be sexual_identity or has this already been agreed with upstream systems?